src/data/sym/sym2.lean

```diff
@@ -230,23 +230,44 @@ lemma eq_of_ne_mem {x y : α} {z z' : sym2 α} (h : x ≠ y)
 @[ext]
 protected lemma ext (z z' : sym2 α) (h : ∀ x, x ∈ z ↔ x ∈ z') : z = z' :=
 begin
-  refine quotient.rec_on_subsingleton z (λ w, _) h,
-  refine quotient.rec_on_subsingleton z' (λ w', _),
-  intro h,
-  cases w with x y, cases w' with x' y',
-  simp only [mem_iff] at h,
-  apply eq_iff.mpr,
+  refine sym2.ind (λ x y, _) z h,
+  refine sym2.ind (λ x' y' h, _) z',
   have hx := h x, have hy := h y, have hx' := h x', have hy' := h y',
-  simp only [true_iff, true_or, eq_self_iff_true, iff_true, or_true] at hx hy hx' hy',
-  cases hx; subst x; cases hy; subst y; cases hx'; try { subst x' }; cases hy'; try { subst y' };
-  simp only [eq_self_iff_true, and_self, or_self, true_or, or_true],
+  simp only [mem_iff, eq_self_iff_true, or_true, iff_true, true_or, true_iff] at hx hy hx' hy',
+  cases hx; cases hy; cases hx'; cases hy'; subst_vars,
+  simp only [sym2.eq_swap],
 end
 
 instance mem.decidable [decidable_eq α] (x : α) (z : sym2 α) : decidable (x ∈ z) :=
 quotient.rec_on_subsingleton z (λ ⟨y₁, y₂⟩, decidable_of_iff' _ mem_iff)
 
 end membership
 
+@[simp] lemma mem_map {f : α → β} {b : β} {z : sym2 α} :
+  b ∈ sym2.map f z ↔ ∃ a, a ∈ z ∧ f a = b :=
+begin
+  refine sym2.ind (λ x y, _) z,
```

Comment:
Does `induction z using sym2.ind with x y` work here? I guess it's longer...

Reply:
Sure, why not. (In fact, back when sym2 was new, I complained about not being able to use `induction` because there is some sort of bug where, I think, it would create the wrong motive if you used `quotient.induction`, but Mario said it likely never will be fixed and just use `refine`. Better respect past me! Thanks.)